core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -1959,7 +1960,17 @@ object ZIOSpec extends ZIOBaseSpec {
       } @@ flaky @@ zioTag(errors),
       test("returns success when it happens after failure") {
         assertZIO(ZIO.fail(42).raceAll(List(ZIO.succeed(24) <* Live.live(ZIO.sleep(100.millis)))))(equalTo(24))
-      } @@ zioTag(errors)
+      } @@ zioTag(errors),
+      test("always calls finalizers") {
+        for {
+          isRunning              <- ZIO.succeed(new AtomicBoolean(true))
+          backgroundBlockingStuff = ZIO.attemptBlocking(while (isRunning.get()) {})
```

Comment:
Sorry I only just noticed this. Can you change this to `ZIO.whileLoop(isRunning.get)(ZIO.yieldNow)(_ => ())` instead and below? Otherwise it might cause issues in low CPU environments due to the background thread being active all the time